### DISCUSSION TITLE
CI feature: Implement pull request preview environments

### DISCUSSION THREAD

**Discussion by waveywaves** [authelia/authelia#4603] [Ideas]
I would like to support Authelia by implementing [Uffizzi](https://github.com/UffizziCloud/uffizzi) preview environments.
Disclaimer: I work on [Uffizzi](https://github.com/UffizziCloud/uffizzi).

Uffizzi is a Open Source full stack previews engine and our platform is available completely free for Authelia (and all open source projects). This will provide maintainers with preview environments of every PR in the cloud, which enables faster iterations and reduces time to merge. You can see the open source repos which are currently using Uffizzi over [here](https://uffizzi.notion.site/)

Uffizzi is purpose-built for the task of previewing PRs and it integrates with your workflow to deploy preview environments in the background without any manual steps for maintainers or contributors. We can go ahead and create an Initial PoC for you right away if you think there is value in this proposal.

- [ ] Initial PoC

**Comment by james-d-elliott** [comment]
Hey thanks for reaching out! We find this idea quite interesting and have a few ways we think this will be beneficial. We'll be in touch.
